namesys/interface.go

```diff
@@ -31,10 +35,7 @@ type NameSystem interface {
 type Resolver interface {
 
 	// Resolve looks up a name, and returns the value previously published.
-	Resolve(ctx context.Context, name string) (value path.Path, err error)
-
-	// CanResolve checks whether this Resolver can resolve a name
-	CanResolve(name string) bool
```

Comment:
On Thu, May 14, 2015 at 09:41:42PM -0700, Juan Batiz-Benet wrote:
> > -	// CanResolve checks whether this Resolver can resolve a name
> > -	CanResolve(name string) bool
> 
> why lose this? it's useful for clients to be able to check this.

Not useful enough for us to use it anywhere ;).  Can you think of a
situation where the local check is useful enough that you'd use it?
I'd just try a depth=1 resolve and see if that errored out.